File: examples/java/operations/parallel/named-branches.java
import software.amazon.lambda.durable.DurableContext;
import software.amazon.lambda.durable.DurableHandler;
import software.amazon.lambda.durable.model.ParallelResult;

public class NamedBranches extends DurableHandler<Void, ParallelResult> {
    @Override
    public ParallelResult handleRequest(Void input, DurableContext context) {
        try (var parallel = context.parallel("process")) {
            parallel.branch("task-a", String.class,
                    ctx -> ctx.step("run-a", String.class, s -> "a done"));
            parallel.branch("task-b", String.class,
                    ctx -> ctx.step("run-b", String.class, s -> "b done"));
            return parallel.get();
        }
    }
}

File: examples/csharp/operations/parallel/named-branches.cs
using Amazon.Lambda.Core;
using Amazon.Lambda.DurableExecution;

public class NamedBranchesExample
{
    public Task<DurableExecutionInvocationOutput> Handler(
        DurableExecutionInvocationInput input, ILambdaContext context)
        => DurableFunction.WrapAsync<object, IReadOnlyList<string>>(Workflow, input, context);

    private async Task<IReadOnlyList<string>> Workflow(object input, IDurableContext ctx)
    {
        // Each DurableBranch carries a name surfaced on IBatchItem.Name.
        IBatchResult<string> result = await ctx.ParallelAsync(
            new[]
            {
                new DurableBranch<string>("task-a", async (branch, ct) =>
                    await branch.StepAsync(async (_, _) => "a done", name: "run-a")),
                new DurableBranch<string>("task-b", async (branch, ct) =>
                    await branch.StepAsync(async (_, _) => "b done", name: "run-b")),
            },
            name: "process");

        return result.GetResults();
    }
}
